File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];

            if (map.containsKey(complement)) { //if we found a pair
                return new int[] { map.get(complement), i };
            }
            map.put(nums[i], i);
        }
        return new int[] {};
    }
}

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map= new Map();

    for(let i=0; i<nums.length; i++){
        const curnum= nums[i];
        const comp= target - curnum;

        if(map.has(comp)){
            return [map.get(comp),i];
        }
        map.set(curnum,i);
    }
    return [];
};